tests/integration/resolve/test_issue_2092.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import os.path
+import sys
+
+import pytest
+
+from pex.cli.testing import run_pex3
+from pex.compatibility import commonpath
+from pex.interpreter import PythonInterpreter
+from pex.pex_info import PexInfo
+from pex.testing import run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+def test_vcs_respects_target(
+    tmpdir,  # type: Any
+    py38,  # type: PythonInterpreter
+    py39,  # type: PythonInterpreter
+):
+    # type: (...) -> None
+
+    lock = os.path.join(str(tmpdir), "lock.json")
+    vcs_requirement = (
+        "emote-rl[torch]@ "
+        "git+https://github.com/EmbarkStudios/emote"
+        "@4c5b31753e7a497fa57ab59e13344468510c920c"
+        "#egg=emote-rl"
+    )
+
+    run_pex3(
+        "lock",
+        "create",
+        "--style",
+        "universal",
+        "--resolver-version",
+        "pip-2020-resolver",
+        "--target-system",
+        "linux",
+        "--target-system",
+        "mac",
+        "--interpreter-constraint",
+        "==3.9.*",
+        vcs_requirement,
+        "-o",
+        lock,
+        "--indent",
+        "2",
+    ).assert_success()
+
+    pex = os.path.join(str(tmpdir), "pex")
+    pex_root = os.path.join(str(tmpdir), "pex_root")
+    run_pex_command(
```

Comment:
This is a direct test of the Slack-reported issue.

With this diff:
```diff
diff --git a/pex/resolve/lock_resolver.py b/pex/resolve/lock_resolver.py
index bbe405f..4cfcb50 100644
--- a/pex/resolve/lock_resolver.py
+++ b/pex/resolve/lock_resolver.py
@@ -112,7 +112,7 @@ class VCSArtifactDownloadManager(DownloadManager[VCSArtifact]):

         requirement = artifact.as_unparsed_requirement(project_name)
         downloaded_vcs = resolver.download(
-            targets=Targets.from_target(self._target),
+            # targets=Targets.from_target(self._target),
             requirements=[requirement],
             transitive=False,
             indexes=self._indexes,
```

You get back this failure, which is exactly as seen in the original issue report.
```console
$ tox -epy311-integration -- -vvsk test_issue_2092 -n0
...
tests/integration/resolve/test_issue_2092.py::test_vcs_respects_target FAILED
...
E         Running command /tmp/pytest-of-jsirois/pytest-13/test_vcs_respects_target0/pex_root/venvs/024cbf2a0ae30fda5f75e87e21d5fc87e10db368/d51ff0ca3b6ad11288a7b24367ef2e6f6c3ee68b/bin/python /tmp/pytest-of-jsirois/pytest-13/test_vcs_respects_target0/pex_root/venvs/024cbf2a0ae30fda5f75e87e21d5fc87e10db368/d51ff0ca3b6ad11288a7b24367ef2e6f6c3ee68b/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/pytest-of-jsirois/pytest-13/test_vcs_respects_target0/pex_root/pip_cache/.tmp/tmpdo_zy3id
E           Running command /tmp/pytest-of-jsirois/pytest-13/test_vcs_respects_target0/pex_root/venvs/024cbf2a0ae30fda5f75e87e21d5fc87e10db368/d51ff0ca3b6ad11288a7b24367ef2e6f6c3ee68b/bin/python /tmp/pytest-of-jsirois/pytest-13/test_vcs_respects_target0/pex_root/venvs/024cbf2a0ae30fda5f75e87e21d5fc87e10db368/d51ff0ca3b6ad11288a7b24367ef2e6f6c3ee68b/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/pytest-of-jsirois/pytest-13/test_vcs_respects_target0/pex_root/pip_cache/.tmp/tmpchkfzmqb
E       ERROR: Package 'emote-rl' requires a different Python: 3.8.10 not in '>=3.9'
...
```